src/interval.jl

```diff
@@ -171,6 +171,38 @@ function Base.last(interval::Interval{T,L,R}) where {T,L,R}
     return R !== Unbounded ? interval.last : nothing
 end
 
+function Base.minimum(interval::Interval{T,Closed,R}; increment=nothing) where {T,R}
+    return first(interval)
+end
+
+function Base.minimum(interval::Interval{T,Open,R}; increment=eps(T)) where {T,R}
+    return first(interval) + increment
+end
+
+function Base.maximum(interval::Interval{T,L,Closed}; increment=nothing) where {T,L}
+    return last(interval)
+end
+
+function Base.maximum(interval::Interval{T,L,Open}; increment=eps(T)) where {T,L}
+    return last(interval) - increment
+end
+
+function Base.minimum(interval::Interval{T,Open,R}; increment=one(T)) where {T<:Integer,R}
+    return first(interval) + increment
+end
+
+function Base.maximum(interval::Interval{T,L,Open}; increment=one(T)) where {T<:Integer,L}
+    return last(interval) - increment
+end
+
+function Base.minimum(interval::Interval{T,Open,R}; increment=eps(T)) where {T<:AbstractFloat,R}
+    return nextfloat(first(interval))
+end
+
+function Base.maximum(interval::Interval{T,L,Open}; increment=eps(T)) where {T<:AbstractFloat,L}
+    return prevfloat(last(interval))
+end
+
```

Comment:
Can we move these definitions to after the `isbounded` function definition? Also, these functions should probably use `AbstractInterval` instead of `Interval` so that this also works for `AnchoredIntervals`.

Reply:
We should also have tests for `AnchoredInterval`s and intervals using `Unbounded`